examples/nvector/kokkos/test_nvector_kokkos.cpp

```diff
@@ -0,0 +1,220 @@
+    VecType Z{X};
+
+    /* Standard vector operation tests */
+    printf("\nTesting standard vector operations:\n\n");
+
+    fails += Test_N_VAbs(X, Z, length, 0);
+    fails += Test_N_VAddConst(X, Z, length, 0);
+    fails += Test_N_VCompare(X, Z, length, 0);
+    fails += Test_N_VConst(X, length, 0);
+    fails += Test_N_VConstrMask(X, Y, Z, length, 0);
+    fails += Test_N_VDiv(X, Y, Z, length, 0);
+    fails += Test_N_VDotProd(X, Y, length, 0);
+    fails += Test_N_VInv(X, Z, length, 0);
+    fails += Test_N_VInvTest(X, Z, length, 0);
+    fails += Test_N_VL1Norm(X, length, 0);
+    fails += Test_N_VLinearSum(X, Y, Z, length, 0);
+    fails += Test_N_VMaxNorm(X, length, 0);
+    fails += Test_N_VMin(X, length, 0);
+    fails += Test_N_VMinQuotient(X, Y, length, 0);
+    fails += Test_N_VProd(X, Y, Z, length, 0);
+    fails += Test_N_VScale(X, Z, length, 0);
+    fails += Test_N_VWL2Norm(X, Y, length, 0);
+    fails += Test_N_VWrmsNorm(X, Y, length, 0);
+    fails += Test_N_VWrmsNormMask(X, Y, Z, length, 0);
+
+    /* Fused and vector array operations tests (disabled) */
+    printf("\nTesting fused and vector array operations (disabled):\n\n");
+
+    /* create vector and test vector array operations */
+    VecType U{X};
+
+    /* fused operations */
+    fails += Test_N_VLinearCombination(U, length, 0);
+    fails += Test_N_VScaleAddMulti(U, length, 0);
+    fails += Test_N_VDotProdMulti(U, length, 0);
+
+    /* vector array operations */
+    fails += Test_N_VLinearSumVectorArray(U, length, 0);
+    fails += Test_N_VScaleVectorArray(U, length, 0);
+    fails += Test_N_VConstVectorArray(U, length, 0);
+    fails += Test_N_VWrmsNormVectorArray(U, length, 0);
+    fails += Test_N_VWrmsNormMaskVectorArray(U, length, 0);
+    fails += Test_N_VScaleAddMultiVectorArray(U, length, 0);
+    fails += Test_N_VLinearCombinationVectorArray(U, length, 0);
+
+    /* local reduction operations */
+    printf("\nTesting local reduction operations:\n\n");
+
+    fails += Test_N_VDotProdLocal(X, Y, length, 0);
+    fails += Test_N_VMaxNormLocal(X, length, 0);
+    fails += Test_N_VMinLocal(X, length, 0);
+    fails += Test_N_VL1NormLocal(X, length, 0);
+    fails += Test_N_VWSqrSumLocal(X, Y, length, 0);
+    fails += Test_N_VWSqrSumMaskLocal(X, Y, Z, length, 0);
+    fails += Test_N_VInvTestLocal(X, Z, length, 0);
+    fails += Test_N_VConstrMaskLocal(X, Y, Z, length, 0);
+    fails += Test_N_VMinQuotientLocal(X, Y, length, 0);
+  }
+  Kokkos::finalize();
+
+  /* Print result */
+  if (fails) { printf("FAIL: NVector module failed %i tests \n\n", fails); }
+  else { printf("SUCCESS: NVector module passed all tests \n\n"); }
+
+  Test_Finalize();
+
+  return (fails);
+}
+
+/* ----------------------------------------------------------------------
+ * Implementation specific utility functions for vector tests
+ * --------------------------------------------------------------------*/
+
+int check_ans(realtype ans, N_Vector X, sunindextype local_length)
```

Comment:
Should this (and the subsequent routines) use `Kokkos::deep_copy`` instead of `sundials::kokkos::CopyFromDevice` and `sundials::kokkos::CopyToDevice`?  Or maybe for maximal coverage, we could switch a subset of these routines to use those, and keep the rest using ours?

Reply:
`CopyFromDevice(v)` and `CopyToDevice(v)` are just wrappers for `Kokkos::deep_copy(v.HostView(), v.View());` and `Kokkos::deep_copy(v.View(), v.HostView());` respectively so I'd prefer use our wrappers here for testing.